amethyst_derive/src/system_desc.rs

```diff
@@ -679,7 +679,8 @@ fn field_computation_expressions(system_desc_fields: &SystemDescFields<'_>) -> T
 
                     let tokens = quote! {
                         let #field_name = world
-                            .fetch_mut::<EventChannel<#event_type_path>>()
+                            .get_mut::<EventChannel<#event_type_path>>()
+                            .unwrap()
```

Comment:
nit pick: can you use:

```rust
let error_message = format!("Expected `EventChannel<{}>` to exist.", quote!(event_type_path).to_string());

// then
.unwrap_or_else(|| panic!(#error_message))
```
(not sure if that's the shortest way to implement that)